index.mjs

```diff
@@ -69,14 +78,22 @@ export function $(pieces, ...args) {
 }
 
 $.verbose = true
-// Try `command`, should cover all Bourne-like shells.
-// Try `which`, should cover most other cases.
-// Try `type` command, if the rest fails.
-$.shell = `${execSync('command -v bash || which bash || type -p bash')}`.trim()
-$.prefix = 'set -euo pipefail;'
-$.quote = shq
 $.cwd = undefined
 
+$.quote = (s) => s
+$.prefix = ''
+if ($.shell = which.sync('bash', { nothrow: true })) {
+  $.prefix = 'set -euo pipefail;'
+  $.quote = shq
+} else if ($.shell = whichSyncFallback(['pwsh', 'powershell'])) {
```

Comment:
Maybe just warn the user about bash no avaialbe?

Reply:
Yeah, that would be a way to go by it. Will do the change.